code/game/objects/items/weapons/grenades/flashbang.dm

```diff
@@ -5,39 +5,42 @@
 	origin_tech = "materials=2;combat=3"
 	light_power = 10
 	light_color = LIGHT_COLOR_WHITE
-	var/light_time = 2
-	var/range = 7
+
+	var/light_time = 0.2 SECONDS // The duration the area is illuminated
+	var/range = 7 // The range in tiles of the flashbang
 
 /obj/item/grenade/flashbang/prime()
 	update_mob()
-	var/flashbang_turf = get_turf(src)
-	if(!flashbang_turf)
+	var/turf/T = get_turf(src)
+	if(!T)
 		return
 
-	set_light(7)
-
+	// VFX and SFX
 	do_sparks(rand(5, 9), FALSE, src)
-	playsound(flashbang_turf, 'sound/effects/bang.ogg', 25, 1)
-	bang(flashbang_turf, src, range)
+	playsound(T, 'sound/effects/bang.ogg', 100, TRUE)
+	new /obj/effect/dummy/lighting_obj(T, light_color, range + 2, light_power, light_time)
 
-	for(var/obj/structure/blob/B in hear(8, flashbang_turf))     		//Blob damage here
-		var/damage = round(30 / (get_dist(B, get_turf(src)) + 1))
-		B.take_damage(damage, BURN, "melee", 0)
+	// Stunning mechanic
+	bang(T, src, range)
 
-	spawn(light_time)
-		qdel(src)
+	// Blob damage here
+	for(var/obj/structure/blob/B in hear(range + 1, T))
+		var/damage = round(30 / (get_dist(B, T) + 1))
+		B.take_damage(damage, BURN, "melee", FALSE)
+
+	qdel(src)
 
 /proc/bang(turf/T, atom/A, range = 7, flash = TRUE, bang = TRUE)
+	var/source_turf = get_turf(A)
 	for(var/mob/living/M in hearers(range, T))
-		if(M.stat == DEAD)
+		if(M.is_dead())
```

Comment:
This is the first time, ever this proc has been used--I admit, I'm a bit skeptical of it, as the only thing it does is check for `stat == DEAD` in and of itself---only with additional proc call overhead.

Reply:
True. Personally I prefer getters and setters where possible (I blame coming from a Java environment for that) but for the sake of consistency with the rest of the code base, I'll probably switch this back to direct value checking.